lib/rails_admin_import/import_model.rb

```diff
@@ -0,0 +1,94 @@
+module RailsAdminImport
+  class AssociationNotFound < StandardError
+  end
+
+  class ImportModel
+    def initialize(abstract_model)
+      @abstract_model = abstract_model
+      @config = abstract_model.config.import
+      @model = abstract_model.model
+    end
+
+    attr_reader :abstract_model, :model, :config
+
+    def display_name
+      abstract_model.config.label
+    end
+
+    def label_for_model(object)
+      object.public_send(label_method)
+    end
+
+    def label_method
+      @label_method ||= abstract_model.config.object_label_method
+    end
+
+    def importable_fields(model_config = config)
+      @importable_fields ||= {}
+      @importable_fields[model_config] ||= model_config.visible_fields.reject do |f|
+        # Exclude id, created_at and updated_at
+        model_config.default_excluded_fields.include? f.name
+      end
+    end
+
+    def model_fields(model_config = config)
+      @model_fields ||= {}
+      @model_fields[model_config] ||= importable_fields(model_config).select { |f|
+        !f.association? || f.association.polymorphic?
+      }
+    end
+
+    def association_fields
+      @association_fields ||= importable_fields.select { |f|
+        f.association? && !f.association.polymorphic?
+      }
+    end
+
+    def belongs_to_fields
+      @belongs_to_fields ||= association_fields.select { |f|
+        !f.multiple?
+      }
+    end
+
+    def many_fields
+      @many_fields ||= association_fields.select {
```

Comment:
Avoid using {...} for multi-line blocks.<br>Pass `&:multiple?` as an argument to `select` instead of a block.